xnmt/translator.py

```diff
@@ -236,12 +237,33 @@ def __call__(self, translator, dec_state, src, trg):
 class TranslatorReinforceLoss(Serializable):
   yaml_tag = '!TranslatorReinforceLoss'
 
+  class Regressor():
+    def __init__(self, dim):
+      self.param_collection = dy.Model()
+      self.trainer = dy.AdamTrainer(self.param_collection)
```

Comment:
Not sure if adding 1 more trainer here is the correct way to incorporate the baseline. See https://github.com/neulab/xnmt/blob/segmenting_encoder/xnmt/segmenting_encoder.py#L211 for example. The idea is to compose the loss of the decoder and baseline together.